cli/internal/cmdutil/cmdutil.go

```diff
@@ -211,8 +216,32 @@ type CmdBase struct {
 }
 
 // LogError prints an error to the UI
-func (b *CmdBase) LogError(format string, args ...interface{}) {
+func (b *CmdBase) LogError(format string, shouldPrintToTerminal bool, args ...interface{}) {
```

Comment:
rather than have a boolean parameter, can we have a different function to do this or separate logging and UI output? Booleans are hard to understand in context, calling `LogError(..., true)` doesn't communicate what the `true` is.

Reply:
Actually, are there cases where we don't want to log errors to the terminal? If we have an error, even in the dry-run case, I think we've failed to produce valid JSON and should display the error.

Reply:
Chatted offline and decided to spot fix and move the log message for Remote Cache so it doesn't run in Dry runs. Everything else remains the same.